src/sage/groups/perm_gps/partn_ref/data_structures.pxd

```diff
@@ -136,6 +136,37 @@ cdef inline void OP_join(OrbitPartition *OP, int m, int n) noexcept:
     if m_root != n_root:
         OP.num_cells -= 1
 
+
+cdef inline void OP_make_set(OrbitPartition *OP) noexcept:
+    cdef int i, n = OP.degree
+    cdef int *new_parent, *new_rank, *new_mcr, *new_size
+
+    cdef int *int_array = <int *> sig_malloc( 4*(n+1) * sizeof(int) )
+    if int_array is NULL:
+        sig_free(int_array)
+    else :
+        OP.degree = n + 1
+        OP.num_cells = OP.num_cells + 1
+        new_parent = int_array
+        new_rank = int_array + (n + 1)
+        new_mcr = int_array + (2*n + 2)
+        new_size = int_array + (3 * n + 3)
+        for i from 0 <= i < n:
```

Comment:
`for i in range(n)`

Reply:
Will that work for c? If you look at all the other for loops in the .pxd file they all have the same format:
`for i from 0 <= i < n:`

Reply:
yes, it will work.

Reply:
K, I've gone ahead and changed it.

For future: is there a reason why all the other ones use `for i from 0 <= i < n`? Because every single other for loop in the file uses this syntax rather than the `for i in range` syntax.

Reply:
These methods have been written long time ago and Cython has evolved (improved) since. We have not update all the code yet, it's too big.